src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Time.txt

```diff
@@ -134,4 +140,203 @@ true
 2
 
 >> Second(Time(1, 2, 3) + Date(2011, 2, 3))
-3
\ No newline at end of file
+3
+
+// *********** OPERATIONS WITH TIMES  *************
+
+>> Time(2,0,0) + Time(2,0,0)
+Time(4,0,0,0)
+
+>> Time(12,50,45,950) + Time(6,15,35,750)
+Time(19,6,21,700)
+
+>> Time(2,0,0) + 0.5
+Time(14,0,0,0)
+
+>> Time(18,0,0) + 0.5
+Time(6,0,0,0)
+
+>> Time(18,0,0) - 0.5
+Time(6,0,0,0)
+
+>> Time(18,0,0) - 0.25
+Time(12,0,0,0)
+
+>> 0.25 + Time(2,0,0)
+Time(8,0,0,0)
+
+>> Time(18,30,0) - Time(6,30,0)
+0.5
+
+>> Date(2022, 9, 8) + Time(12, 34, 56)
+DateTime(2022,9,8,12,34,56,0)
+
+>> Time(12, 34, 56) + Date(2022, 9, 8)
+DateTime(2022,9,8,12,34,56,0)
+
+>> DateTime(2022, 9, 8, 1, 1, 1, 1) + Time(12, 34, 56)
+DateTime(2022,9,8,13,35,57,1)
+
+>> Time(12, 34, 56) + DateTime(2022, 9, 8, 1, 1, 1, 1)
+DateTime(2022,9,8,13,35,57,1)
+
+>> Date(2022,1,1) + 2.5
+DateTime(2022,1,3,12,0,0,0)
+
+>> 2.5 + Date(2022,1,1)
+DateTime(2022,1,3,12,0,0,0)
+
+>> Time(12,0,0) - Date(2022,10,10)
+Errors: Error 13-14: This operation isn't valid on these types: Time, Date.
+
+>> Time(12,0,0) - DateTime(2022,10,10,0,0,0)
+Errors: Error 13-14: This operation isn't valid on these types: Time, DateTime.
+
+>> 0 - Time(2,30,45)
+#Error(Kind=NotSupported)
+
+// *********** COMPONENTS FROM OVERFLOWS **********
+>> Hour(Time(26,90,90))
+3
+
+>> Minute(Time(26,90,90))
+31
+
+>> Second(Time(26,90,90,5500))
+35
+
+>> Hour(DateTime(2000,15,40,26,90,90))
+3
+
+>> Minute(DateTime(2000,15,40,26,90,90))
+31
+
+>> Second(DateTime(2000,15,40,26,90,90,5500))
+35
+
+// ********** COMPONENTS FROM UNDERFLOWS **********
+>> Hour(Time(4,-90,-90))
+2
+
+>> Minute(Time(4,-90,-90))
+28
+
+>> Second(Time(4,-90,-90))
+30
+
+>> Hour(DateTime(2000,15,10,-4,-300,-300,-30000))
+14
+
+>> Minute(DateTime(2000,15,10,-4,-300,-300,-30000))
+54
+
+>> Second(DateTime(2000,15,10,-4,-300,-300,-30000))
+30
+
+// ************** BLANK AND ERRORS ****************
+>> Time(Blank(), 34, 56)
+Time(0,34,56,0)
+
+>> Time(12, Blank(), 56)
+Time(12,0,56,0)
+
+>> Time(12, 34, Blank())
+Time(12,34,0,0)
+
+>> Time(12, 34, 56, Blank())
+Time(12,34,56,0)
+
+>> Hour(If(1<0,Time(1,1,1)))
+0
+
+>> Hour(If(1<0,DateTime(2000,1,1,1,1,1)))
+0
+
+>> Minute(If(1<0,Time(1,1,1)))
+0
+
+>> Minute(If(1<0,DateTime(2000,1,1,1,1,1)))
+0
+
+>> Second(If(1<0,Time(1,1,1)))
+0
+
+>> Second(If(1<0,DateTime(2000,1,1,1,1,1)))
+0
+
+>> Time(1/0, 34, 56)
+#Error(Kind=Div0)
+
+>> Time(12, Sqrt(-1), 56)
```

Comment:
>Time(12, Sqrt(-1), 56) [](http://example.com/codeflow?start=3&length=22)

That should be Time(12, i, 56), no? :)

Reply:
We don't have to worry about this - at least not yet. Do you want to be the one to introduce complex numbers to Power Fx? :-D